app/js/streaming/StreamController.js

```diff
@@ -298,12 +298,33 @@
             self.notify(MediaPlayer.dependencies.StreamController.eventList.ENAME_STREAMS_COMPOSED);
         },
 
+        onTimeSyncAttemptCompleted = function (e) {
+            var self = this,
+                manifest;
+
+            if (!e.error) {
+                // adjust the loadedTime of the manifest by the offset between
+                // the network time and the local time. this is because loadedTime
+                // is used in a number of places to work out availability etc.
+                manifest = self.manifestModel.getValue();
```

Comment:
This concerns me. The current approach is to keep client wall-clock times and then use an offset during all calculations. The offset is the output of the binary search rule. This should just set the same offset that the binary search rule would. Any cases where that offset is not taken into account need to be fixed.